R/visualizations.R

```diff
@@ -173,10 +178,13 @@ plotScoreHeatmap <- function(results, cells.use = NULL, labels.use = NULL,
 
     args <- list(mat = scores[,order,drop=FALSE], border_color = NA, show_colnames = FALSE,
         clustering_method = 'ward.D2', cluster_cols = cluster_cols, ...)
-
-    if (!is.null(clusters)) {
-        args$annotation_col <- clusters[order,,drop=FALSE]
-    }
+    
+    #Set Annotations
```

Comment:
This section is not entirely safe if someone didn't use the full `annotation_col` argument name. Better to restore `annotation_col` as a full argument to this function. 

Also, `scores` may have been `order`ed, so indexing/naming by `colnames(scores)` in 184 won't be correct if the existing `annotation_col` is meant to respect the original ordering of cells in tihe input `scores`.

The book-keeping interactions between `annotation_col`, `scores`, `order` and `cells.use` (if `annotation_cols` is supplied), is quite tricky. I would suggest you assemble `annotation_cols` as you go along (replacing it with `NULL` at the end if it has no columns) rather than leaving it all to the end.

Also, why do people insist on using `&` and not `&&`? The latter is the correct way to do it. See:

```r
x <- NA
if (!is.na(x) & y > 0) { print("YAY") } # Errors
if (!is.na(x) && y > 0) { print("YAY") } # No error, as second condition doesn't evaluate if the first fails.
```

Reply:
All good points.  To the && question... for me, it's because I didn't know!  Oh the code simplifications I can now make in DittoSeq too... thanks!
For the potential mismatch of 'annotation_col' with 'scores'/'order'/'cells.use', it helps that pheatmap utilizes the row.names to match annotations to columns.  I can use these to ensure proper alignment with 'Clusters' and 'Pruned' annotations as well.

- [x] Add annotation_col to docs
- [ ] Add names to the clustering and prune.calls before any reordering
- [ ] Use names to ensure proper alignment and subsetting of annotations going into the final heatmap.  (pheatmap utilizes row.names to do the matching)
- [ ] && instead of &.

Reply:
Done? (sending to trigger a notification because I'm pretty sure comment edits to not!)